### ISSUE TITLE
Riverpod should depend on `state_notifier` >= 0.7.2

### ISSUE THREAD

**Issue by ElteHupkes** [rrousselGit/riverpod#1903]
**Describe the bug**
After updating to any Riverpod 2.x, from 1.x, I get the following error while building:

```
.............

Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/x1/r1hmz0m918j148w80zpxjv_c0000gn/T/flutter_tools.viLFl6/flutter_ios_build_temp_dirab0eJA/temporary_xcresult_bundle

    ../../bin/flutter/.pub-cache/hosted/pub.dartlang.org/riverpod-2.1.1/lib/src/state_notifier_provider/base.dart:158:10: Error: The method 'updateShouldNotify' isn't defined for the class 'StateNotifier<T>'.
     - 'StateNotifier' is from 'package:state_notifier/state_notifier.dart' ('../../bin/flutter/.pub-cache/hosted/pub.dartlang.org/state_notifier-0.7.1/lib/state_notifier.dart').
    Try correcting the name to the name of an existing method, or defining a method named 'updateShouldNotify'.
            .updateShouldNotify(previous, next);
             ^^^^^^^^^^^^^^^^^^

.............
```

A short investigation reveals that Riverpod [depends on `state_notifier: ^0.7.0`](https://github.com/rrousselGit/riverpod/blob/master/packages/riverpod/pubspec.yaml#L16), however, `updateShouldNotify()` [isn't present](https://github.com/rrousselGit/state_notifier/blob/v0.7.0/packages/state_notifier/lib/state_notifier.dart) until [version 0.7.2](https://github.com/rrousselGit/state_notifier/blob/v0.7.2/packages/state_notifier/lib/state_notifier.dart). 

Having installed the original dependencies a while ago, I have `state_notifier: "0.7.1"` in my `pubspec.lock`. Since it matches the dependency, it's not updated by `pub get`. 

**To Reproduce**

In a project with Riverpod installed, update `pubspec.lock` to include an older version of `state_notifier` that still matches the spec (0.7.0 or 0.7.1), clean the project, run `pub get`, build.

**Expected behavior**
The `state_notifier` dependency should be updated to the minimum required version when updating Riverpod.

**Workaround for people affected**
Two options:
- Manually update `pubspec.lock` to have `state_notifier: "0.7.2+1"`, clean, run `pub get`
- Just delete `pubspec.lock` altogether and clean, run `pub get`, that should also get the latest supported versions of dependencies in most cases.

**Comment by rrousselGit** [comment]
Run `pub run upgrade`

**Comment by ElteHupkes** [comment]
Thanks, that would be another workaround; I've already fixed it locally.

All of the above was a quite verbose (sorry, figured somebody might Google the error and end up here) way to say that riverpod depends on `state_notifier: ^0.7.0` but won't actually build with `0.7.0`. I assume that if the `pubspec.yaml` of the packages would have `state_notifier: ^0.7.2` instead, the update would've been performed automatically when I updated riverpod.
